caluma/caluma_core/tests/test_visibility.py

```diff
@@ -167,6 +167,42 @@ class ConfiguredUnion(Union):
     assert queryset.count() == 0
 
 
+def test_union_visibility_return_managers(db, history_mock):
+    FakeModel = get_fake_model(
+        dict(name=CharField(max_length=255)), model_base=models.UUIDModel
+    )
+    FakeModel.objects.create(name="Name1")
+    FakeModel.objects.create(name="Name2")
+    FakeModel.objects.create(name="Name3")
+    FakeModel.objects.create(name="Name4")
+
+    class CustomNode(DjangoObjectType):
+        class Meta:
+            model = FakeModel
+
+    class Name1Visibility(BaseVisibility):
+        @filter_queryset_for(Node)
+        def filter_queryset_for_node(self, node, queryset, info):
+            return queryset
+
+    class Name2Visibility(BaseVisibility):
```

Comment:
Not sure having two visibility classes has any impact on the aspect being tested

Reply:
It has: `.union()` is not called on the first, hence the bug is not triggered.

Reply:
Sorry, it's the opposite: `.union()` is called on the first with the second one as argument. And that triggers the bug.